core/src/main/kotlin/org/evomaster/core/search/gene/ObjectGene.kt

```diff
@@ -187,25 +187,36 @@ class ObjectGene(
         return additionalFields!!.any { it.first.value == fieldToAdd.first.value}
     }
 
-    override fun copyValueFrom(other: Gene) {
+    override fun copyValueFrom(other: Gene): Boolean {
         if (other !is ObjectGene) {
             throw IllegalArgumentException("Invalid gene type ${other.javaClass}")
         }
 
         if (other.isFixed != isFixed)
             throw IllegalArgumentException("cannot copy value for ObjectGene if their isFixed is different")
 
+        val current = copy()
+
+        var ok = true
+
         for (i in fixedFields.indices) {
-            this.fixedFields[i].copyValueFrom(other.fixedFields[i])
+            ok = ok && this.fixedFields[i].copyValueFrom(other.fixedFields[i])
```

Comment:
can't this lead to a partial update? eg, the first N fields get copied, and then one is wrong, which leads to return `false` with a partial update?

Reply:
it might lead to the partial update, but it will be fixed with `copyValueFrom(current)` in the following code.